pkgs/development/compilers/ghc/8.10.7.nix

```diff
@@ -212,6 +212,14 @@ stdenv.mkDerivation (rec {
       url = "https://gitlab.haskell.org/ghc/ghc/-/commit/97d0b0a367e4c6a52a17c3299439ac7de129da24.patch";
       sha256 = "0r4zjj0bv1x1m2dgxp3adsf2xkr94fjnyj1igsivd9ilbs5ja0b5";
     })
+  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [
```

Comment:
Unsure if this should be optional. In general optional patches are quite a nightmare and easily lead to regresions not being noticed, but since this patch is *quite* invasive and also pretty hacky, I'm inclined to leave it conditional.

Not sure what the other haskell maintainers think.

Reply:
Technically, I suppose the patch isn't just applicable to `arm64-darwin` but all other platforms; It doesn't have to be an optional patch. But I do agree, this is a hack around the fact that DCE is broken with LLVM.

Reply:
Yeah, I dunno. aarch64 darwin has been such a pain, but I don't know if that's really an argument for isolating any changes there.

Reply:
Since this has suboptimal error messages (just warnings in a different place from where the compilation happens, unnecessary warnings if the symbols are unused) and doesn't fix anything for non `aarch64-darwin`, I would like to keep this conditional.